sway-lsp/src/capabilities/document_symbol.rs

```diff
@@ -21,30 +32,113 @@ pub fn to_symbol_information(tokens: &[Token], url: Url) -> Vec<SymbolInformatio
+                Declaration::TraitDeclaration(_) => SymbolKind::INTERFACE,
+                Declaration::StructDeclaration(_) => SymbolKind::STRUCT,
+                Declaration::EnumDeclaration(_) => SymbolKind::ENUM,
+                Declaration::ConstantDeclaration(_) => SymbolKind::CONSTANT,
+                Declaration::ImplTrait { .. } => SymbolKind::INTERFACE,
+                Declaration::AbiDeclaration(_) => SymbolKind::INTERFACE,
+                // currently we return `variable` type as default
+                Declaration::Reassignment(_)
+                | Declaration::ImplSelf { .. }
+                | Declaration::StorageDeclaration(_)
+                | Declaration::Break { .. }
+                | Declaration::Continue { .. } => SymbolKind::VARIABLE,
+            }
+        }
+        AstToken::Expression(exp) => {
+            match &exp {
+                Expression::Literal { value, .. } => match value {
+                    Literal::String(_) => SymbolKind::STRING,
+                    Literal::Boolean(_) => SymbolKind::BOOLEAN,
+                    _ => SymbolKind::NUMBER,
+                },
+                Expression::FunctionApplication { .. } => SymbolKind::FUNCTION,
+                Expression::VariableExpression { .. } => SymbolKind::VARIABLE,
+                Expression::Array { .. } => SymbolKind::ARRAY,
+                Expression::StructExpression { .. } => SymbolKind::STRUCT,
+                // currently we return `variable` type as default
+                _ => SymbolKind::VARIABLE,
+            }
+        }
+        AstToken::FunctionDeclaration(_) => SymbolKind::FUNCTION,
+        AstToken::FunctionParameter(_) => SymbolKind::TYPE_PARAMETER,
+        AstToken::StructField(_) => SymbolKind::FIELD,
+        AstToken::EnumVariant(_) => SymbolKind::ENUM_MEMBER,
+        AstToken::TraitFn(_) => SymbolKind::FUNCTION,
+        AstToken::StorageField(_) => SymbolKind::FIELD,
+        AstToken::Reassignment(_) => SymbolKind::VARIABLE,
+    }
+}
+
+fn typed_to_symbol_kind(typed_ast_token: &TypedAstToken) -> SymbolKind {
+    match typed_ast_token {
+        TypedAstToken::TypedDeclaration(dec) => {
+            match dec {
+                TypedDeclaration::VariableDeclaration(_) => SymbolKind::VARIABLE,
+                TypedDeclaration::ConstantDeclaration(_) => SymbolKind::CONSTANT,
+                TypedDeclaration::FunctionDeclaration(_) => SymbolKind::FUNCTION,
+                TypedDeclaration::TraitDeclaration(_) => SymbolKind::INTERFACE,
+                TypedDeclaration::StructDeclaration(_) => SymbolKind::STRUCT,
+                TypedDeclaration::EnumDeclaration(_) => SymbolKind::ENUM,
+                TypedDeclaration::ImplTrait { .. } => SymbolKind::INTERFACE,
+                TypedDeclaration::AbiDeclaration(_) => SymbolKind::INTERFACE,
+                TypedDeclaration::GenericTypeForFunctionScope { .. } => SymbolKind::TYPE_PARAMETER,
+                // currently we return `variable` type as default
```

Comment:
Weird, is there no keyword symbol kind or anything long those lines? Or should we possibly be returning `None` or similar? Not a blocker as I can see this was the original behaviour, just curious.

Reply:
Na there isn't a keyword variant. I agree though that we should scrutinize exactly the role and mappings of nodes to SymbolKind's in a future PR. I was just keeping the original behavior in place for now.